client/src/App.js

```diff
@@ -1,11 +1,135 @@
+import React, { useState, useEffect } from "react";
 import "./App.css";
+import Videocard from "./Videocard";
+import AddingItem from "./AddingItem";
+import { ErrorBoundary } from "react-error-boundary";
 
 function App() {
+  const [videos, setVideos] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [updated, setUpdated] = useState(true);
+  const [sortOrder, setSortOrder] = useState("desc");
+
+  useEffect(() => {
+    fetch("https://videos-1rb3.onrender.com/")
+      .then((response) => response.json())
+      .then((data) => {
+        setVideos(data);
+        setLoading(false);
+    })
+      .catch((error) => console.log(error));
+  }, [updated]);
+
+  const addVideo = (newVideo) => {
+    fetch("https://videos-1rb3.onrender.com/", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newVideo),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log("Response:", data);
+        setUpdated(!updated);
+      })
+      .catch((error) => {
+        console.error("Error:", error);
+      });
+  };
+
+  const upVote = (video) => {
+    const updatedVideos = videos.map((v) =>
```

Comment:
Nice logic that only change the vote of a particular video.

FYI do you know the difference between `by reference` and `by value`?
https://stackoverflow.com/a/6605700